source/module_elecstate/module_charge/charge_mixing.cpp

```diff
@@ -896,10 +1002,13 @@ double Charge_Mixing::inner_product_recip(std::complex<double>* rho1, std::compl
 double Charge_Mixing::inner_product_recip_new1(std::complex<double>* rho1, std::complex<double>* rho2)
 {
     double rnorm = 0.0;
+    // consider a resize for mixing_angle
+    int resize_tmp = 1;
+    if (GlobalV::NSPIN == 4 && GlobalV::MIXING_ANGLE > 0) resize_tmp = 2;
```

Comment:
I hope you can replace the use of "GlobalV" with local variables or class members in Charge_Mixing code.